### PR TITLE
Allow sharing/duplicating activities/templates

### PR DESCRIPTION
**Pull request by houshuang**

### CODE REVIEW

File: frog/imports/client/SingleActivity/ConfigPanel.js [hunk 1]
```diff
@@ -21,9 +21,11 @@ import { style } from './style';
  */
 class ConfigPanel extends React.Component<
   {
+    name: string,
     activityType: ActivityPackageT,
     onSubmit: Function,
-    onReturn: Function
+    onReturn: Function,
+    data?: Object
```
Comment (qgolsteyn) [inline frog/imports/client/SingleActivity/ConfigPanel.js:28]: What is the format of data?

File: frog/imports/client/SingleActivity/Finish.js [hunk 2]
```diff
@@ -40,14 +39,20 @@ function Finish(
...
         You're all set, please share this link with the participants:{' '}
-        {BASE_URL + '/' + url.public}
+        {BASE_URL + '/' + slug}
       </Typography>
       <Typography variant="h4" component="h3" className={classes.padded_text}>
         Here's a secret link to your behind-the-scene view of the dashboard:{' '}
-        <Link to={'/' + url.dashboard + '?u=' + Meteor.userId()}>
-          {BASE_URL + '/' + url.dashboard + '?u=' + Meteor.userId()}
+        <Link to={'/t/' + slug + '?u=' + Meteor.userId()}>
+          {BASE_URL + '/t/' + slug + '?u=' + Meteor.userId()}
```
Comment (qgolsteyn) [inline frog/imports/client/SingleActivity/Finish.js:47]: We should move `/t` to a constant to simplify future work. A change to the path could break this link.

File: frog/imports/client/SingleActivity/index.js [hunk 3]
```diff
@@ -9,20 +10,38 @@ import ConfigPanel from './ConfigPanel';
...
+          activityTypes.find(x => x.id === data.activityType) ||
+          templatesObj[data.activityType];
+        this.setState({
+          stage: 2,
+          activityType: type,
+          config: data.config
+        });
+      }
+    });
+  };
```
Comment (qgolsteyn) [inline frog/imports/client/SingleActivity/index.js:45]: We should isolate this call outside of `index.js` to limit the number of responsibilities SingleActivity has.

**Review by qgolsteyn — CHANGES_REQUESTED** [review]
Left some comments

File: frog/imports/client/SingleActivity/ConfigPanel.js (hunk 1 above)
Comment (houshuang) [reply frog/imports/client/SingleActivity/ConfigPanel.js:28]: This is the result of react-jsonschema-form for operators/activities, so Object is the most specific we can be.

File: frog/imports/client/SingleActivity/Finish.js (hunk 2 above)
Comment (houshuang) [reply frog/imports/client/SingleActivity/Finish.js:47]: Let's add another task to think about constants and URLs. Where should those constants be defined and exported from?

**Comment by houshuang** [comment]
@qgolsteyn I've tried to address your comments, OK to merge?

**Review by qgolsteyn — APPROVED** [review]
LGTM :+1:

**Pull request merged**